File: CCC-school-work/comporg/pnc2/ktodd3/go/primeregb.go
package main
import "fmt"
import "os"
import "time"
import "strconv"

func main() {
var numberofprimes int
var err error

numberofprimes, err = strconv.Atoi(os.Args[1])
testednumber:=2
n:=0

if err != nil {
os.Exit(0)
}

start := time.Now()

for n < numberofprimes {
	m:=2
	for m<=testednumber{
		if m == testednumber{
			n++
			fmt.Fprintf(os.Stderr,"%d ", testednumber)
		}
		if testednumber%m==0 && testednumber !=2{
			break
		}
		m++
	}
	testednumber++
}
end := time.Now()
diff := end.Sub(start)
fmt.Fprintf(os.Stdout, "\n%8.4f\n", diff.Seconds())
}

File: CCC-school-work/comporg/pnc2/ktodd3/primeregb.rb

numberofprimes = ARGV[0].to_i


testnumber = 2
test = 0
n = 0

start = Time.now

	while n<numberofprimes do
	m = 2
		while m <= testnumber do
			if m == testnumber
				n += 1 
#				$stderr.puts "%d " % testnumber
				$stderr.printf "%d " % testnumber
			elsif testnumber % m == 0 && testnumber != 2
				break
			end
			m+=1
		end
	testnumber += 1
	end
puts
finish = Time.now
delta = finish - start
puts "%8.4f\n" % delta
